### PR TITLE
Feature/improvements

### PR DESCRIPTION
**Pull request by JakeBar**
Changes made:

* add pipenv
* add circle ci and pre-commit
* make font slightly smaller
* Optional QR codes (default is false)
* hide label overflow
* update copyright from 2015 -> 2019

**This PR contains breaking changes**

### CODE REVIEW

**Comment by JakeBar** [comment]
Sample test code:

```
from fpdf import FPDF
import pkg_resources

class Label:
    def __init__(self, name):
        self.name = name

class Card:
    def __init__(self):
        self.name = 'Department menu being cut off again after adding Kogan Super'
        self.labels = [
            Label('Content'),
            Label('Product Catalogue'),
            Label('Issue in Production'),
            Label('Design'),
            Label('Kogan Insurance'),
            Label('Kogan Internet'),
            Label('Kogan Mobile'),
            Label('Kogan Super'),
            Label('PRINTED'),
            Label('Maintenance'),
        ]

def generate_pdf(card):
    """
    Make a PDF from a card

    :param card: dict from fetcher.py
    :return: Binary PDF buffer
    """
    from eclaire.base import SPECIAL_LABELS, MAX_LABEL_CHARS

    pdf = FPDF("L", "mm", (62, 140))
    pdf.set_margins(2.8, 2.8, 2.8)
    pdf.set_auto_page_break(False, margin=0)

    pdf.add_page()

    font = pkg_resources.resource_filename("eclaire", "font/Clairifont.ttf")
    pdf.add_font("Clairifont", fname=font, uni=True)
    pdf.set_font("Clairifont", size=44)

    pdf.multi_cell(0, 18, txt=card.name.upper(), align="L")

    # May we never speak of this again.
    pdf.set_fill_color(255, 255, 255)
    pdf.rect(0, 55, 140, 20, 'F')

    pdf.set_font("Clairifont", "", 16)
    pdf.set_y(-4)
    labels = ", ".join([label.name for label in card.labels if label.name not in SPECIAL_LABELS])[
        :MAX_LABEL_CHARS
    ]
    pdf.multi_cell(0, 0, labels, 0, "R")
    pdf.output('sample.pdf', 'F')

card = Card()

generate_pdf(card)
```

**Comment by JakeBar** [comment]
[sample.pdf](https://github.com/kogan/eClaire/files/3557857/sample.pdf)

File: authors.txt [hunk 1]
```diff
@@ -12,3 +12,4 @@ Michael Cooper
 Nam Ngo
 Sean Kozer
 Simon Knox
+Jake Barber
```
Comment (amcgavin) [inline authors.txt]: >Original authors prior to public release (in alphabetical order):

Neither prior to public release or in alphabetical order 😉

File: eclaire/base.py [hunk 2]
```diff
@@ -1,45 +1,28 @@
-# Copyright 2015 KOGAN.COM PTY LTD
```
Comment (amcgavin) [inline eclaire/base.py:1]: Why are you removing the copyright? this is a public repo.

File: eclaire/base.py [hunk 3]
```diff
@@ -114,16 +93,15 @@ def remove_label(self, card, label):
         At the time of writing there is no way to remove a label with py-trello
         """
         self.trello_client.fetch_json(
-            '/cards/' + card.id + '/idLabels/' + label.id,
-            http_method="DELETE"
+            "/cards/" + card.id + "/idLabels/" + label.id, http_method="DELETE"
```
Comment (amcgavin) [inline eclaire/base.py]: please fix with a `"".format()`

File: eclaire/main.py [hunk 4]
```diff
@@ -1,30 +1,14 @@
 #!/usr/bin/env python
 
-# Copyright 2015 KOGAN.COM PTY LTD
```
Comment (amcgavin) [inline eclaire/main.py:3]: bring back

File: eclaire/notifications.py [hunk 5]
```diff
@@ -1,56 +1,37 @@
-# Copyright 2015 KOGAN.COM PTY LTD
```
Comment (amcgavin) [inline eclaire/notifications.py:1]: copyright

File: eclaire/render.py [hunk 6]
```diff
@@ -1,26 +1,9 @@
-# Copyright 2015 KOGAN.COM PTY LTD
```
Comment (amcgavin) [inline eclaire/render.py:1]: copyright

File: setup.py [hunk 7]
```diff
@@ -1,43 +1,19 @@
-# Copyright 2015 KOGAN.COM PTY LTD
```
Comment (amcgavin) [inline setup.py:1]: copyright

File: setup.py [hunk 8]
```diff
@@ -1,43 +1,19 @@
...
-
-from setuptools import setup, find_packages
-
+from setuptools import find_packages, setup
 
 setup(
-    name='eClaire',
-    version='1.0.0',
+    name="eClaire",
+    version="1.0.0",
```
Comment (amcgavin) [inline setup.py]: bump version, this will be a breaking release as qr code is being removed.

File: eclaire/render.py [hunk 9]
```diff
@@ -56,49 +36,29 @@ def generate_pdf(card):
...
-    font = pkg_resources.resource_filename('eclaire', 'font/Clairifont.ttf')
-    pdf.add_font('Clairifont', fname=font, uni=True)
-    pdf.set_font('Clairifont', size=48)
-
-    pdf.multi_cell(0, 18, txt=card.name.upper(), align='L')
+    font = pkg_resources.resource_filename("eclaire", "font/Clairifont.ttf")
+    pdf.add_font("Clairifont", fname=font, uni=True)
+    pdf.set_font("Clairifont", size=48)
 
-    qrcode = generate_qr_code(card.shortUrl)
```
Comment (amcgavin) [inline eclaire/render.py:73]: lets make printing qrcode a configurable option?

**Review by amcgavin — CHANGES_REQUESTED** [review]

File: eclaire/base.py (hunk 2 above)
Comment (JakeBar) [reply eclaire/base.py:1]: There's already an apache licence in the base directory. I don't think this is required per file

Comment (JakeBar) [reply eclaire/base.py:1]: That's my assumption

Comment (amcgavin) [reply eclaire/base.py:1]: tbh I don't know either

File: eclaire/render.py (hunk 9 above)
Comment (JakeBar) [reply eclaire/render.py:73]: Sure can

File: eclaire/base.py (hunk 2 above)
Comment (JakeBar) [reply eclaire/base.py:1]: > Each original source document (code and documentation, but excluding the LICENSE and NOTICE files) SHOULD include a short license header at the top. If the distribution contains documents not covered by CLA, CCLA or Software Grant (such as third-party libraries) then see the policy guide

http://www.apache.org/dev/apply-license.html

I'll bring it back

File: eclaire/render.py (hunk 9 above)
Comment (simonwillcock) [reply eclaire/render.py:73]: oo nice change

**Pull request merged**